libarchive/archive_read_support_format_rar.c

```diff
@@ -972,7 +972,7 @@ archive_read_format_rar_read_header(struct archive_read *a,
     case MAIN_HEAD:
       rar->main_flags = archive_le16dec(p + 3);
       skip = archive_le16dec(p + 5);
-      if (skip < 7 + sizeof(rar->reserved1) + sizeof(rar->reserved2)) {
+      if ((size_t)skip < 7 + sizeof(rar->reserved1) + sizeof(rar->reserved2)) {
```

Comment:
(This is actually the reason I asked about the type of `skip`, but I see now that it has no risk of truncation or overflow after conversion here since it was just filled and is guaranteed to be `0 <= x <= 65535`)